src/character/player.rs

```diff
@@ -1,56 +1,252 @@
 use bevy::{math::vec3, prelude::*};
+use bevy_replicon::{
+    client::ClientSet,
+    network_event::{
+        client_event::{ClientEventAppExt, FromClient},
+        server_event::{SendMode, ServerEventAppExt, ToClients},
+        EventType,
+    },
+    replicon_core::replication_rules::{AppReplicationExt, Ignored},
+    server::{ServerSet, SERVER_ID},
+};
+use serde::{Deserialize, Serialize};
+
+use crate::network::{
+    client::Client, has_client, has_client_and_local, has_local_player, has_server,
+    replication::transform::SyncedTransform,
+};
 
 use super::{CharacterPhysicsBundle, CharacterVectors};
 
 pub const RADIUS: f32 = 0.4;
 pub const HALF_HEIGHT: f32 = 0.4;
 
-#[derive(Component)]
-pub struct PlayerControls;
+pub struct PlayerPlugin;
+
+impl Plugin for PlayerPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_server_event::<TransformEvent>(EventType::Ordered)
+            .add_client_event::<TransformEvent>(EventType::Ordered)
+            .replicate::<Player>()
+            .add_systems(
+                PreUpdate,
+                (
+                    init_players.after(ClientSet::Receive).run_if(has_client()),
+                    transform_server_handler
+                        .after(ServerSet::Receive)
+                        .run_if(has_server()),
+                    transform_client_handler
+                        .after(ClientSet::Receive)
+                        .run_if(has_client()),
+                ),
+            )
+            .add_systems(
+                FixedUpdate,
+                (
+                    control.run_if(has_local_player()),
+                    transform_server_sender.run_if(has_server()),
+                    transform_client_sender.run_if(has_client_and_local()),
+                ),
+            );
+    }
+}
+
+pub fn spawn(
+    commands: &mut Commands,
+    meshes: &mut ResMut<Assets<Mesh>>,
+    materials: &mut ResMut<Assets<StandardMaterial>>,
+    player: Player,
+    is_local: bool,
```

Comment:
Please do not use `bool` type arguments in functions. Prefer an `enum` instead, which is a lot more readable.
```rs
spawn(commands, meshes, materials, player, true); // what does the true mean?
spawn(commands, meshes, materials, player, PlayerKind::Local); // much better!
```